flashinfer/kda_kernels/fused_kda_decode_multitoken.py

```
@@ -0,0 +1,1054 @@
+    cute.arch.barrier()
+
+    # ---- L2 norms of q, k and their cross term: one warp per token, each
+    # ---- lane owning 4 channels, so the whole 128-wide reduction is a single
+    # ---- butterfly and the token loop below reads 3 ready-made scalars
+    # ---- instead of re-reducing 12 partials per token.
+    a_qkv = sQKV.iterator.toint()
+    for nb in cutlass.range_constexpr((T * 32 + NT - 1) // NT):
+        idx = nb * NT + tid
+        if idx < T * 32:
+            t = idx // 32
+            l = idx % 32
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)